### DISCUSSION TITLE
Update Date field on change

### DISCUSSION THREAD

**Discussion by renrom** [Budibase/budibase#9381] [Help] [answered]
Can't get this to work properly. I have a date field, if i fill the date, i want to fill in another datefield with one year difference.

Somehow, the calculation take's place on the previous values..

this is the first screen:

![image](https://user-images.githubusercontent.com/26928569/213267245-c21042b4-907b-40fe-a26c-f498e123757d.png)

This is the screen after selecting a date:

![image](https://user-images.githubusercontent.com/26928569/213267378-7d1d17ae-a4cb-4a31-b49c-67eb42b2122d.png)

And this after change the first field again:

![image](https://user-images.githubusercontent.com/26928569/213267477-806a84b0-5c72-4ff4-969b-f3fb9fcebfc9.png)

I tried the following:

On Change --> Update Field Value: Value:

Tried it with Form.Fields, but also with Form.Value. same result.

![image](https://user-images.githubusercontent.com/26928569/213267682-2f076e13-38fd-4099-816b-136e3f218ba9.png)

Also tried it with Set State, same result.

Is this not possible what i ask?

**Comment by andz-bb** [comment, marked as answer]
hey @renrom 

Your original solution is very close to being correct! What you are trying to do is possible by using the `Field Value` binding instead of the value from the Form Field. Here's an example of the JS binding to use:

```
const start = new Date($("Field Value"));

const jaar = start.getFullYear() + 1;

const nieuwedatum = new Date(start.setFullYear(jaar));

return nieuwedatum;
```

This will get the new value from the change event itself so it will always be up to date with the change you've just made.

![field value on change](https://user-images.githubusercontent.com/110921612/213442663-16c47fca-2216-428b-975f-d353e0171dcb.gif)

Hope this helps!

**Reply by renrom** [reply]
Yes! that was it! Thanks a lot!.  I am pretty new with budibase, and i do struggle a bit with finding the right variable to use. Could i found this somewhere in the documentation? or is this common sense in javascript :)

**Reply by andz-bb** [reply]
@renrom glad it helped! There's a mention of it in the docs here: https://docs.budibase.com/docs/actions#on-change Is there anything I could do to make it clearer or easier to find?

**Reply by renrom** [reply]
@andz-bb , i did find the video's very helpfull! The possibilities with budibase is huge.. i did manage to make an inventory app with qr code generation and list with items that are almost overdue. I could not imagine making this with any other app actually.. and i did look at a lot of other apps. 

i do think its my lag of experiance with javascript/handlebars to get the full potention out of it, i do am getting there though.. not in 5 minutes haha..

Next steps with this app is email integration and sending (selected) data with it (if its possible though). and alerts if some threshholds are reach.. 

Small steps :D

**Comment by NicolasFabre81** [comment]
Hello ! 
I'm having a similar problem : I do use the "Field value" binding , but I have several fields to update in a row.
When I update the first one and press TAB, the "on change" event fires 2 times ! One for the first field, one for the second !
So this cause to have the first field updated value to be lost some times...
Any workaround, except waiting for the new grid component ?
<img width="424" alt="image" src="https://github.com/Budibase/budibase/assets/28592069/c4a92056-577b-40d1-9a57-efb6a7a0c502">

EDIT : Instead of using TAB, if after the update I click anywhere to quit the first field, then it's OK : on change fires only 1 time !

EDIT2 : Found a first solution : add a CONTINUE IF to the second field :
<img width="385" alt="image" src="https://github.com/Budibase/budibase/assets/28592069/0781c57e-b63e-437f-9e47-80872e877f1a">
But not perfect because the second field looses the focus.

@lgasteceau
